src/rpc/mining.cpp

```diff
@@ -375,7 +375,7 @@ static RPCHelpMan generateblock()
         LOCK(cs_main);
 
         BlockValidationState state;
-        if (!TestBlockValidity(state, chainparams, block, LookupBlockIndex(block.hashPrevBlock), false, false)) {
+        if (!TestBlockValidity(state, chainparams, ::ChainstateActive(), block, g_chainman.m_blockman.LookupBlockIndex(block.hashPrevBlock), false, false)) {
```

Comment:
Any reason not to just jump straight to using `EnsureChainman(request.context)` in all the rpc methods that you're touching here?

Reply:
I don't think it makes too much of a difference, and I found it clearer to push that decision to a later commit when we've resolved everything under RPC and can see everything that depends on chainman. See: 02e3831e9cd10f78bfde2d67a99c47f08564f07b

Reply:
ok!